### DISCUSSION TITLE
How to use Coverlet with Cake 3.0?

### DISCUSSION THREAD

**Discussion by taconaut** [cake-build/cake#4072] [Q&A] [answered]
Hi,

Until now (up to Cake 2.3.0), I've been using executing my tests like this, to include coverage results using coverlet:
```
#addin nuget:?package=Cake.Coverlet&version=2.5.4

Task("Unit-Test")
    .IsDependentOn("Application-Build")
    .Does (() => {
        RunTests(GetFiles($"./**/{_applicationBaseName}*.Tests.Unit.csproj"));
    });

var _testSettings = new DotNetCoreTestSettings {
    NoBuild = true,
    NoRestore = true,
    Configuration = _configuration,
    ResultsDirectory = _testResultsDir
};

var _coverletSettings = new CoverletSettings {
    CollectCoverage = true,
    CoverletOutputFormat = CoverletOutputFormat.cobertura,
    CoverletOutputDirectory = _testCoverageResultsDir,
    Exclude = new List<string> { "[xunit*]*", $"[{_applicationBaseName}.Tests.*]*" },
    Include = new List<string> { $"[{_applicationBaseName}*]*" }
};

private void RunTests(FilePathCollection projectFiles){
    foreach(var projectFile in projectFiles)
    {
        var projectFilePath = projectFile.Segments.Last();

        // Test result customization
        var testResultFileName = $"test-results-{projectFilePath}.xml";
        _testSettings.ArgumentCustomization = args => args.Append($"--logger:trx;LogFileName={testResultFileName}");

        // Coverage result customization
        var coverageResultFileName = $"coverage-results-{projectFilePath}.cobertura.xml";
        _coverletSettings.CoverletOutputName = coverageResultFileName;

        DotNetCoreTest(projectFile.FullPath, _testSettings, _coverletSettings);
    }
}
```

Now, if keeping
```
var _testSettings = new DotNetCoreTestSettings {
    NoBuild = true,
    NoRestore = true,
    Configuration = _configuration,
    ResultsDirectory = _testResultsDir
};

var _coverletSettings = new CoverletSettings {
    CollectCoverage = true,
    CoverletOutputFormat = CoverletOutputFormat.cobertura,
    CoverletOutputDirectory = _testCoverageResultsDir,
    Exclude = new List<string> { "[xunit*]*", $"[{_applicationBaseName}.Tests.*]*" },
    Include = new List<string> { $"[{_applicationBaseName}*]*" }
};

private void RunTests(FilePathCollection projectFiles){
    foreach(var projectFile in projectFiles)
    {
        var projectFilePath = projectFile.Segments.Last();

        // Test result customization
        var testResultFileName = $"test-results-{projectFilePath}.xml";
        _testSettings.ArgumentCustomization = args => args.Append($"--logger:trx;LogFileName={testResultFileName}");

        // Coverage result customization
        var coverageResultFileName = $"coverage-results-{projectFilePath}.cobertura.xml";
        _coverletSettings.CoverletOutputName = coverageResultFileName;

        DotNetCoreTest(projectFile.FullPath, _testSettings, _coverletSettings);
    }
}
```
I get:
> D:/Dev/Hci.MedicalData.AppInfo/Source/build.cake(188,25): error CS0246: The type or namespace name 'DotNetCoreTestSettings' could not be found (are you missing a using directive or 
an assembly reference?)
D:/Dev/Hci.MedicalData.AppInfo/Source/build.cake(195,29): error CS0246: The type or namespace name 'CoverletSettings' could not be found (are you missing a using directive or an assembly reference?)
D:/Dev/Hci.MedicalData.AppInfo/Source/build.cake(197,28): error CS0103: The name 'CoverletOutputFormat' does not exist in the current context
D:/Dev/Hci.MedicalData.AppInfo/Source/build.cake(216,9): error CS0103: The name 'DotNetCoreTest' does not exist in the current context
Error: Error(s) occurred when compiling build script:
D:/Dev/Hci.MedicalData.AppInfo/Source/build.cake(188,25): error CS0246: The type or namespace name 'DotNetCoreTestSettings' could not be found (are you missing a using directive or 
an assembly reference?)
D:/Dev/Hci.MedicalData.AppInfo/Source/build.cake(195,29): error CS0246: The type or namespace name 'CoverletSettings' could not be found (are you missing a using directive or an assembly reference?)
D:/Dev/Hci.MedicalData.AppInfo/Source/build.cake(197,28): error CS0103: The name 'CoverletOutputFormat' does not exist in the current context
D:/Dev/Hci.MedicalData.AppInfo/Source/build.cake(216,9): error CS0103: The name 'DotNetCoreTest' does not exist in the current context

When using the renamed classes (without 'Core' in the name):
```
var _testSettings = new DotNetTestSettings {
    NoBuild = true,
    NoRestore = true,
    Configuration = _configuration,
    ResultsDirectory = _testResultsDir
};
private void RunTests(FilePathCollection projectFiles){
    foreach(var projectFile in projectFiles)
    {
        var projectFilePath = projectFile.Segments.Last();

        // Test result customization
        var testResultFileName = $"test-results-{projectFilePath}.xml";
        _testSettings.ArgumentCustomization = args => args.Append($"--logger:trx;LogFileName={testResultFileName}");

        // Coverage result customization
        var coverageResultFileName = $"coverage-results-{projectFilePath}.cobertura.xml";
        _coverletSettings.CoverletOutputName = coverageResultFileName;

        DotNetTest(projectFile.FullPath, _testSettings);
    }
}
```
I can't pass the coverlet settings to `DotNetTest` (as coverlet hasn't been updated and doesn't seem to be maintained anymore).

Have I got a way to make this work as before?
```

**Comment by devlead** [comment, marked as answer]
DotNetCore* aliases were deprecated and obsoleted with Cake 2.0 a little over a year ago, as they've were replaced by DotNet* and removed with Cake 3.0 last week. 

In a Cake script "DotNetCore" could be found and replaced with "DotNet" and thing should just work, if an addin references 1.0 it would need to reference 3.0, do the same find and replace and ship a new version to nuget.

More details in our release blog post
https://cakebuild.net/blog/2022/11/cake-v3.0.0-released

**Reply by taconaut** [reply]
Thanks, inquiring here how to proceed https://github.com/Romanx/Cake.Coverlet/pull/50
